charts/istio-alerts/templates/service-prometheusrule.yaml

```diff
@@ -85,5 +82,36 @@ spec:
         {{ toYaml . | nindent 8}}
         {{- end }}
     {{- end }}
-  {{- end }}
+    {{- end }}
+    {{- with .Values.serviceRules.destinationServiceSelectorValidity }}
+    {{- if .enabled }}
+    - alert: DestinationServiceSelectorValidity
+      annotations:
+        summary: >-
+          DestinationServiceSelector for istio-alerts is invalid
+        runbook_url: {{ $values.defaults.runbookUrl}}#Alert-Rules-Selectors-Validity
+        description: >-
+          The DestinationServiceSelector used for service level alerts did not return any data.
+          Please check the DestinationServiceSelector applied in your istio-alerts chart
+          is correct to ensure you are properly selecting your service so that you
+          will be alerted for service issues. The current selector is
+          `{destination_service_namespace="{{ $release.Namespace }}", destination_service_name="{{ $.Values.serviceRules.destinationServiceName }}"}`.
```

Comment:
Can we replace these hard-coded lists of the selectors with a variable - that way you can just refer to `$istioSelector`? We started doing that at https://github.com/Nextdoor/k8s-charts/blob/main/charts/prometheus-alerts/templates/containers-prometheusrule.yaml#L2-L8 and https://github.com/Nextdoor/k8s-charts/blob/main/charts/prometheus-alerts/templates/_helpers.tpl#L13-L59 and it makes the code more readable and easier to modify later on.

Reply:
I actually want to split that into a follow up PR as I do not want to conflate the code change of adding a new metric with changing the code of the existing metrics.